std/xml.d

```diff
@@ -562,7 +562,7 @@ class Document : Element
         string tagString = xml.tag.tagString;
 
         this(xml.tag);
-        prolog = s[0 .. tagString.ptr - s.ptr];
+        prolog = s[0 .. &tagString[0] - &s[0]];
```

Comment:
Sorry, I still don't get it.  If tagString is empty, wouldn't it be illegal to take the address of its (nonexistent) first element? Or is the whole point to trigger a druntime bounds check upon taking the address?